alonet/torch2trt/base_exporter.py

```diff
@@ -147,15 +157,62 @@ def build_torch_model(self):
         pass
         raise Exception("Child class should implement this method")
 
-    def adapt_graph(self, graph):
+
+
+    def adapt_graph(self, graph: gs.Graph):
         """Modify ONNX graph to ensure compability between ONNX and TensorRT
 
         Returns
         -------
         graph: onnx_graphsurgeon.Graph
         """
-        pass
-        raise Exception("Child class should implement this method")
+        return graph
+
+    def _adapt_graph(self, graph):
+        """Modify ONNX graph to ensure compability between ONNX and TensorRT
+
+        Returns
+        -------
+        graph: onnx_graphsurgeon.Graph
+        """
+        clip_nodes = get_nodes_by_op("Clip", graph)
+        def handle_op_Clip(node: gs.Node):
+            max_constant = np.array(np.finfo(np.float32).max, dtype=np.float32)
+            if "value" in node.inputs[1].i().inputs[0].attrs:
+                min_constant = node.inputs[1].i().inputs[0].attrs["value"].values.astype(np.float32)
+                if len(node.inputs[2].inputs) > 0:
+                    max_constant = node.inputs[2].i().inputs[0].attrs["value"].values.astype(np.float32)
+            elif "to" in node.inputs[1].i().inputs[0].attrs:
+                min_constant = np.array(np.finfo(np.float32).min, dtype=np.float32)
+            else:
+                raise Exception("Error")
+            node.inputs.pop(1)
+            node.inputs.insert(1, gs.Constant(name=node.name + "_min", values=min_constant))
+            node.inputs.pop(2)
+            node.inputs.insert(2, gs.Constant(name=node.name + "_max", values=max_constant))
+
+        for n in clip_nodes:
+            handle_op_Clip(n)
+
+        from onnxsim import simplify
+        model = onnx.load(self.onnx_path)
+        check = False
+        model_simp, check = simplify(model)
+
+        if check:
+            print("\n[INFO] Simplified ONNX model validated. Graph optimized...")
+            graph = gs.import_onnx(model_simp)
+            graph.toposort()
+            graph.cleanup()
+        else:
+            print("\n[INFO] ONNX model was not validated.")
+
+        if self.use_scope_names:  # Rename nodes to correct profiling
```

Comment:
`rename_nodes_` should maybe be called directly in `_torch2onnx` before saving the graph, to keep all the renaming operations in one place.